controller/src/main/java/org/jboss/as/controller/access/JmxAction.java

```diff
@@ -47,11 +65,42 @@ public Impact getImpact() {
 
     /**
      * Gets the {@link javax.management.MBeanServer} method name that was called
+     * @return
      */
     public String getMethodName() {
         return methodName;
     }
 
+    public String getOperation() {
+        return operation;
+    }
+
+    public boolean hasAttribute() {
+        return attribute != null;
+    }
+
+    public String getAttributeName() {
+        return attribute;
+    }
+
+    public ModelNode getCurrentValue() {
+        return currentValue;
+    }
```

Comment:
I'm not sure we should have this method if we can't properly populate the value. (AFAICT it's always 'null' now.)

In any case, it should return Object. We have no mechanism to consistently convert the arbitrary types JMX supports to ModelNode. The only use of this now is SensitiveVaultExpressionConstraint, and that code can do an "instanceof String" check and if true wrap the string in a ModelNode for the isSensitiveValue check.